### PR TITLE
Document why `str.to_{lower,upper}case` return `String`

### LINKED ISSUE #39201: Document why str.to_lowercase and str.to_uppercase return a String
I haven't checked, but I'm 95% certain that they do this because some characters expand into multiple characters when casefolding which wouldn't fit into the buffer of the `str`, so we need a new buffer. Since we don't know the size beforehand, we just return a `String`.

Probably should link to char.to_lowercase and char.to_uppercase.

### PR DESCRIPTION
**Pull request by tbu-**
Fixes #39201.

### CODE REVIEW

**Comment by tbu-** [comment]
CC @Havvy

File: src/libcollections/str.rs [hunk 1]
```diff
@@ -1646,6 +1646,10 @@ impl str {
     /// 'Lowercase' is defined according to the terms of the Unicode Derived Core Property
     /// `Lowercase`.
     ///
+    /// Since some characters can expand into multiple when changing the case,
```
Comment (Havvy) [inline src/libcollections/str.rs]: I would say "multiple characters" instead of "multiple".

File: src/libcollections/str.rs [hunk 2]
```diff
@@ -1718,6 +1722,10 @@ impl str {
     /// 'Uppercase' is defined according to the terms of the Unicode Derived Core Property
     /// `Uppercase`.
     ///
+    /// Since some characters can expand into multiple when changing the case,
```
Comment (Havvy) [inline src/libcollections/str.rs]: Likewise here.

**Review by Havvy — CHANGES_REQUESTED** [review]

File: src/libcollections/str.rs (hunk 1 above)
Comment (tbu-) [inline src/libcollections/str.rs]: Fixed.

**Comment by steveklabnik** [comment]
@bors: r+ rollup

thanks!

**Pull request merged**